src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -668,8 +668,16 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
       this.legend = layerParams['legend'] ? layerParams['legend'] : null;
 
       if (!this.evalscript) {
-        this.evalscript = layerParams['evalscript'] ? layerParams['evalscript'] : null;
+        const dataProductUrl = layerParams['dataProduct'];
+        this.evalscript = layerParams['evalscript'];
+        if (!this.evalscript && dataProductUrl) {
```

Comment:
I think checking if evalscript is present and fetching the evalscript could be directly in `fetchLayerParamsFromSHServiceV3()` which is also called in `getMap()` if `evalscript` or `dataProduct` are not defined. This would in some cases eliminate the need for usage of `layer.updateLayerFromServiceIfNeeded()` in the apps.

Reply:
Moved, thanks!